server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -166,6 +167,17 @@ public DocumentMapperParser documentMapperParser() {
         return this.documentParser;
     }
 
+    public FieldType getLuceneFieldType(String field) {
```

Comment:
This is a bit of an unfortunate hack - there are still places that require access directly to the lucene field type, particularly in the highlighter code.  They can almost certainly be refactored to use either information from the MappedFieldType or from the Mapper, but this PR is big enough already.

Reply:
++.